server/src/server.js

```diff
@@ -1065,15 +1073,16 @@ const END_POINT_SEPARATOR = "|";
 
 // Get OAuth client params for the client-side authorization flow.
 app.get("/api/fxa-oauth/login/*", async function(req, res, next) {
-  if (!req.deviceId) {
-    next(errors.missingSession());
-    return;
-  }
-
   try {
     const stateBytes = await randomBytes(32);
     let state = stateBytes.toString("hex");
-    const inserted =  await setState(req.deviceId, state);
+    const cookies = new Cookies(req, res, {keys: dbschema.getKeygrip("auth")});
+    const stateId = req.deviceId || String(Math.random());
```

Comment:
We should generate the stateId with a UUID instead of Math.random() (I think this was left over from my extra hacky beginning work)

Reply:
done